### ISSUE TITLE
lambdasWithApiDependency error when resource does not have a dependsOn

### ISSUE THREAD

**Issue by gitandrewk** [aws-amplify/amplify-cli#7361]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v12.14.0

### Amplify CLI Version

4.51.1

### What operating system are you using?

Windows

### Amplify Categories

auth, storage, function, api, analytics, hosting, notifications

### Amplify Commands

push

### Describe the bug

After trying to remove a GraphQL model, experiencing a ```Cannot read property 'find' of undefined``` when trying to push. Appears to be due to my cloudformation not having a "dependsOn".

### Expected behavior

Should be able to remove a GraphQL model and push successfully.

### Reproduction steps

1. Create GraphQL model
2. Create functions
3. Amplify Push
4. Remove a GraphQL model
5. Amplify Push
6. Error

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

I was able to "fix" by modifying the lambdasWithApiDependency in ```
amplify-cli/packages/amplify-category-function/src/provider-utils/awscloudformation/utils/getDependentFunction.js``` and wrapping in a try/catch, but suspect it could also be changed to 
```
# Line 15
const lambdaFuncResources = allResources.filter(
            resource =>
              resource.service === ServiceName.LambdaFunction &&
              resource.mobileHubMigrated !== true &&
              resource.dependsOn !== undefined &&
              resource.dependsOn.find(val =&gt; val.category === 'api'),
);
```

**Comment by ammarkarachi** [comment]
@gitandrewk Is your function given access to the data model and then you modified the schema to remove that specific model?

**Comment by gitandrewk** [comment]
I believe it was, but I was receiving the error message before prompting that I needed to remove the model dependency. Unfortunately, I was not able to test to see if removing the dependency first would have given the error message.

Also, I tried reverting my `lambdasWithApiDependency` and removed another model, but did not receive an error message.

Apologies that I don't have more details - I was making too many changes to capture more use cases.
